raster/src/main/scala/geotrellis/raster/imagery/CloudRemoval.scala

```diff
@@ -0,0 +1,127 @@
+package geotrellis.raster.imagery
+
+import geotrellis.raster._
+import geotrellis.raster.render._
+import geotrellis.raster.io.geotiff.SingleBandGeoTiff
+
+import java.io.File
+import spire.syntax.cfor._
+
+object Imagery {
+
+  def cloudlessValue(images: Array[Tile], col: Int, row: Int, threshold: Int): Int = {
+    var sum = 0
+    var count = 0
+    cfor(0)(_ < images.length, _ + 1) { i =>
+      val v = images(i).get(col, row)
+      if(isData(v) && v < threshold) {
+        sum += v
+        count += 1
+      }
+    }
+    sum/count
+  }
+
+  def cloudRemovalSingleBand(images: Array[Tile]) : Tile = {
+    val dummyTile = images(0)
+    val threshold = 10000
+
+    dummyTile.map((col, row, x) => cloudlessValue(images, col, row, threshold))
+  }
+
+  def cloudRemovalMultiBand(images: Array[MultiBandTile]): MultiBandTile = {
+
+    val numBands = images(0).bandCount
+    val numImages = images.length
+
+    val cloudlessTiles = new Array[Tile](numBands)
+
+    cfor(0)(i => i < numBands, i => i + 1) { i =>
+      val singleTiles = new Array[Tile](numImages)
+
+      cfor(0)(j => j < numImages, j => j + 1) { j =>
+        singleTiles(j) = images(j).band(i)
+      }
+      cloudlessTiles(i) = cloudRemovalSingleBand(singleTiles)
+    }
+    ArrayMultiBandTile(cloudlessTiles)
+  }
+
+  def resampleToByte(t: Tile, minVal: Int, maxVal: Int): Tile = {
+    val byteTile = ByteArrayTile.empty(t.cols, t.rows)
+    t.foreach { (col, row, z) =>
+      val v = if(isData(z))
+        ((z - minVal).toDouble/maxVal)*255
+      else 0
+
+      byteTile.set(col, row, v.toInt)
+    }
+    byteTile
+  }
+
+  def writePng(t: MultiBandTile, filename: String) = {
```

Comment:
This will need to be generalized (by just removing the `cloudless` nomenclature) and moved to `geotrellis.raster.render`. We should have this as part of an implicit method that is added to MultiBandTile. It should be called `renderPng`, and create a `PNG` object. The user can write the PNG to a file with the `write` call if so desired. Something like https://github.com/geotrellis/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/render/RenderMethods.scala should be created (say `MultiBandRenderMethods.scala`) to hold this method, and it should use this technique https://github.com/geotrellis/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/render/package.scala#L4 to implicitly add the method to MultiBandTile whenever someone does `import geotrellis.raster.render._`